pkg/util/util.go

```diff
@@ -51,30 +52,33 @@ func MarshalCloudConfig(cloudConfig map[string]runtime.RawExtension) ([]byte, er
 		return []byte{}, nil
 	}
 
-	var err error
-	bytes := []byte("#cloud-config\n")
-
-	for k, v := range cloudConfig {
-		var jsonData []byte
-		if jsonData, err = v.MarshalJSON(); err != nil {
-			return nil, fmt.Errorf("%s: %w", k, err)
-		}
+	// This creates a parent "root" key to facilitate parsing the schemaless map
+	mapSlice := yaml.JSONObjectToYAMLObject(map[string]interface{}{"root": cloudConfig})
+	if len(mapSlice) <= 0 {
+		return nil, errors.New("Could not convert json cloudConfig object to yaml")
+	}
 
-		var structData interface{}
-		if err := json.Unmarshal(jsonData, &structData); err != nil {
-			log.Debugf("failed to decode %s (%s): %s", k, string(jsonData), err.Error())
-			return nil, fmt.Errorf("%s: %w", k, err)
-		}
+	// Just marshal the value of the "root" key
+	yamlData, err := yaml.Marshal(mapSlice[0].Value)
+	if err != nil {
+		return nil, fmt.Errorf("marshalling yaml: %w", err)
+	}
 
-		var yamlData []byte
-		if yamlData, err = yaml.Marshal(structData); err != nil {
-			return nil, err
+	// Determine whether this is a yip config or a cloud-init one.
+	// Since all fields are optional in yip, we test whether any stage has been defined,
+	// as an indication that this is indeed a yip config.
+	yipConfig := &schema.YipConfig{}
```

Comment:
Given we are embracing schema adoption in server side I am wondering if should we get full benefit of it. We making stronger the coupling between elemental-operator and elemental-toolkit through yip, so I am wondering if shouldn't we consider strict schema use. I am, providing a validation on the schema to detect wrong syntax on server side.
We could consider adapting yip for a strict validation. Then here we could simply try yip syntax  and on error try cloud-init syntax and on error simply error out and make it visible through the resource state something went wrong. I think this would be a great improvement to prevent non supported cloud-init syntax. What you think?

Reply:
Would indeed be nice to have validation. Right now indeed you could write anything in there and it will fail during registration/install, which is way too late.

This however seem to suggest we need a validation webhook, so we should first implement that, then rebase this PR to it (plus the strict validation as you suggested)